Source/Core/AudioCommon/Mixer.cpp

```diff
@@ -130,19 +130,30 @@ unsigned int CMixer::Mix(short* samples, unsigned int num_samples)
 
   memset(samples, 0, num_samples * 2 * sizeof(short));
 
-  const bool stretch = SConfig::GetInstance().m_audio_stretch;
+  if (SConfig::GetInstance().m_audio_stretch)
+  {
+    unsigned int actual_samples = std::min({
+        m_dma_mixer.AvailableSamples(), m_streaming_mixer.AvailableSamples(), num_samples,
```

Comment:
No need to handle num_samples here, just alloc a temporary buffer.
Else the dma_mixer might run out of space...